src/web/WebLogService.cpp

```diff
@@ -110,6 +110,10 @@ WebLogService::QueuedLogMessage::QueuedLogMessage(unsigned long id, std::shared_
 }
 
 void WebLogService::operator<<(std::shared_ptr<uuid::log::Message> message) {
+    if (ESP.getMaxAllocHeap() < 20480) {
```

Comment:
i was was always worried these ESP.() functions are expensive and sometimes blocking, but if it works its good!